bdsx/bds/blockpos.ts

```diff
@@ -103,89 +103,87 @@ export class BlockPos extends NativeStruct {
         }
     }
 
-    relative(facing:Facing, steps:number):BlockPos {
+    relative(facing: Facing, steps: number): BlockPos {
         abstract();
     }
 
     static create(pos: Vec3): BlockPos;
     static create(pos: BlockPos): BlockPos;
-    static create(pos: {x:number, y:number, z:number}): BlockPos;
-    static create(x:number, y:number, z:number):BlockPos;
-    static create(a:number|{x:number, y:number, z:number}, b?:number, c?:number):BlockPos {
-        const v = new BlockPos(true);
-        if(typeof a === "number") {
-            v.x = a;
-            v.y = b!;
-            v.z = c!;
+    static create(pos: { x: number, y: number, z: number }): BlockPos;
+    static create(x: number, y: number, z: number): BlockPos;
+    static create(a: number | { x: number, y: number, z: number }, b: number = 0, c: number = 0): BlockPos {
+        if (typeof a === "number") {
+            const v = new BlockPos(true);
+            v.x = a > 0 ? Math.ceil(a) : Math.floor(a);
```

Comment:
Could you tell me why it ceils at the positive number or floors at the negative number?

Reply:
Because Math.round(-20.5) = -21, the correct position of the block is -20

Reply:
What is it correct for?
by the way, `Math.floor(-20.5)` is also `-21`.
it's okay for using `a|0` if you want to round toward zero.
but I think both are not useful for this case.

Reply:
I was wrong, Math.round(-12.5) = -12 and -12.5 | 0 = -12 but correct blockpos is -13. Math.round(12.5) = 13, 12.5 | 0 = 12, correct blockpos is 12.

Reply:
a|0 is only true for positive numbers

Reply:
than `Math.ceil(12.5)` is 13.
it seems you just want to floor.

Reply:
Oh, that's crazy, what was I thinking :v